### ISSUE TITLE
cifar10 result not good as expect !

### ISSUE THREAD

**Issue by zyoohv** [martinarjovsky/WassersteinGAN#62]
I run your code in cifar10, but the result seems not as good as our expected.

* system information:

system: debian 8  
python: python2
pytorch: torch==0.3.1

* run command:

```
$python main.py --dataset cifar10 --dataroot ~/.torch/datasets --cuda 
```

* output part:

```
[24/25][735/782][3335] Loss_D: -1.287177 Loss_G: 0.642245 Loss_D_real: -0.651701 Loss_D_fake 0.635477
[24/25][740/782][3336] Loss_D: -1.269792 Loss_G: 0.621307 Loss_D_real: -0.657210 Loss_D_fake 0.612582
[24/25][745/782][3337] Loss_D: -1.250543 Loss_G: 0.636843 Loss_D_real: -0.667046 Loss_D_fake 0.583497
[24/25][750/782][3338] Loss_D: -1.196252 Loss_G: 0.589907 Loss_D_real: -0.606480 Loss_D_fake 0.589772
[24/25][755/782][3339] Loss_D: -1.189609 Loss_G: 0.564263 Loss_D_real: -0.612895 Loss_D_fake 0.576714
[24/25][760/782][3340] Loss_D: -1.178156 Loss_G: 0.586755 Loss_D_real: -0.600268 Loss_D_fake 0.577888
[24/25][765/782][3341] Loss_D: -1.087157 Loss_G: 0.508717 Loss_D_real: -0.522565 Loss_D_fake 0.564592
[24/25][770/782][3342] Loss_D: -1.092081 Loss_G: 0.674212 Loss_D_real: -0.657483 Loss_D_fake 0.434598
[24/25][775/782][3343] Loss_D: -0.937950 Loss_G: 0.209016 Loss_D_real: -0.310877 Loss_D_fake 0.627073
[24/25][780/782][3344] Loss_D: -1.316574 Loss_G: 0.653665 Loss_D_real: -0.693675 Loss_D_fake 0.622899
[24/25][782/782][3345] Loss_D: -1.222763 Loss_G: 0.558372 Loss_D_real: -0.567426 Loss_D_fake 0.655337
```
fake_samples_500.png
![fake_samples_500](https://user-images.githubusercontent.com/16134679/45865905-9a46ea80-bdb1-11e8-99c5-7ee2c8432cf6.png)

fake_samples_1000.png
![fake_samples_1000](https://user-images.githubusercontent.com/16134679/45865910-9c10ae00-bdb1-11e8-8158-acc3f2e42146.png)

fake_samples_1500.png
![fake_samples_1500](https://user-images.githubusercontent.com/16134679/45865913-9dda7180-bdb1-11e8-9a41-0cd490c124f7.png)

fake_samples_2000.png
![fake_samples_2000](https://user-images.githubusercontent.com/16134679/45865917-9f0b9e80-bdb1-11e8-8f02-a4ddb21c2f85.png)

fake_samples_2500.png
![fake_samples_2500](https://user-images.githubusercontent.com/16134679/45865921-a0d56200-bdb1-11e8-9e2c-101644e49cda.png)

fake_samples_3000.png
![fake_samples_3000](https://user-images.githubusercontent.com/16134679/45865924-a29f2580-bdb1-11e8-9a03-f61b328b2d87.png)

Note that this is real_samples.png!!!
![real_samples](https://user-images.githubusercontent.com/16134679/45865928-a468e900-bdb1-11e8-8c26-5eca7b41bdfa.png)

**Comment by martinarjovsky** [comment]
The samples you’re showing seem to be from super early in training (3000
iterations). Do you have samples from later on?

**Comment by zyoohv** [comment]
The running of the code has finished !

**Comment by zyoohv** [comment]
@martinarjovsky 

I train it 25000 iters, but the result seems until not right.
Could you help me find out what's wrong with it?

**Comment by zyoohv** [comment]
@martinarjovsky 

fake_samples_25000.png
![fake_samples_25000](https://user-images.githubusercontent.com/16134679/45871846-04b35700-bdc1-11e8-9fd1-5f93bcd0acb7.png)


loss log
```
[167/1000][555/782][24987] Loss_D: -0.772816 Loss_G: -0.022373 Loss_D_real: -0.193230 Loss_D_fake 0.579586
[167/1000][560/782][24988] Loss_D: -0.800285 Loss_G: 0.586990 Loss_D_real: -0.633111 Loss_D_fake 0.167174
[167/1000][565/782][24989] Loss_D: -0.585662 Loss_G: 0.091040 Loss_D_real: -0.018860 Loss_D_fake 0.566802
[167/1000][570/782][24990] Loss_D: -0.930666 Loss_G: 0.580418 Loss_D_real: -0.650177 Loss_D_fake 0.280490
[167/1000][575/782][24991] Loss_D: -0.745919 Loss_G: 0.111690 Loss_D_real: -0.156507 Loss_D_fake 0.589412
[167/1000][580/782][24992] Loss_D: -0.981289 Loss_G: 0.589674 Loss_D_real: -0.631602 Loss_D_fake 0.349687
[167/1000][585/782][24993] Loss_D: -0.933379 Loss_G: 0.301309 Loss_D_real: -0.388805 Loss_D_fake 0.544573
[167/1000][590/782][24994] Loss_D: -1.077024 Loss_G: 0.548679 Loss_D_real: -0.589278 Loss_D_fake 0.487745
[167/1000][595/782][24995] Loss_D: -0.914252 Loss_G: 0.511773 Loss_D_real: -0.556782 Loss_D_fake 0.357470
[167/1000][600/782][24996] Loss_D: -1.090694 Loss_G: 0.532181 Loss_D_real: -0.572747 Loss_D_fake 0.517948
[167/1000][605/782][24997] Loss_D: -0.898265 Loss_G: 0.501241 Loss_D_real: -0.532952 Loss_D_fake 0.365313
[167/1000][610/782][24998] Loss_D: -0.943638 Loss_G: 0.485056 Loss_D_real: -0.502485 Loss_D_fake 0.441153
[167/1000][615/782][24999] Loss_D: -0.991872 Loss_G: 0.501545 Loss_D_real: -0.539097 Loss_D_fake 0.452775
[167/1000][620/782][25000] Loss_D: -1.001911 Loss_G: 0.499365 Loss_D_real: -0.527643 Loss_D_fake 0.474269
```

I train it 25000 iters, but the result seems until not right.
Could you help me find out what's wrong with it?

**Comment by Samarth-Tripathi** [comment]
I have the same problem. Maybe its the issue with the new pytorch version? This warning specifically?

UserWarning: volatile was removed and now has no effect. Use `with torch.no_grad():` instead.
  noisev = Variable(noise, volatile = True) # totally freeze netG

**Comment by praveenkumarchandaliya** [comment]
I have change the model into 256 image size(Input image size 64 to 256 size).
then run this code 41600 iteration (800 epochs and 270 iterations at Batchsize 32).
I have use 9000 face image data set.
but over result is not generated good.
/home/mnit/PycharmProjects/ICB2019/WGAN_Pytorch_Clf256/samples/fake_samples_41600.png
![fake_samples_41600](https://user-images.githubusercontent.com/27194129/47269729-a4f6ca00-d57f-11e8-9dc5-45939601b339.png)

**Comment by Mercurial1101** [comment]
> @praveenkumarchandaliya
> 
> I think you can try small images such as 32_32 or 64_64. The method work well in all dataset with small image size in my experiment.
> 
> good luck.



> @praveenkumarchandaliya
> 
> I think you can try small images such as 32_32 or 64_64. The method work well in all dataset with small image size in my experiment.
> 
> good luck.

Have you got good results using CIFAR10 with default parameters? How many epochs have you run? Thanks!

**Comment by martinarjovsky** [comment]
I haven't run the code in cifar 10. You may want to take a look at https://github.com/igul222/improved_wgan_training where we provide a very good cifar10 model.

Cheers :)
Martin
